### PR TITLE
Auto-compile linenoise on build

### PR DESCRIPTION
**Pull request by wilzbach**
Quite a simple idea to use the `preBuildCommands` hooks, but this should work on most Posix platforms as `gcc` is pretty universal. Of course, this isn't perfect and won't work on Windows.

- [uses git subtree](https://www.atlassian.com/blog/git/alternatives-to-git-submodule-git-subtree) as DUB doesn't fetch git submodules on `dub fetch`
- Ideally DUB should provide such abstractions / build layers (e.g. https://github.com/dlang/dub/issues/852, https://github.com/dlang/dub/wiki/DEP5)

### CODE REVIEW

**Comment by wilzbach** [comment]
Sorry for the commit noise, but there were some issues with the way I checked in the module with `git subtree` - it is resolved now & I also removed the `linenoise` installation from Travis to test that the following will work for any Posix system:

```
dub fetch drepl && dub run drepl
```

(and maybe just [run soon](https://github.com/dlang/dub/pull/1082)).

**Comment by wilzbach** [comment]
CC @MartinNowak @ZombineDev - what's your opinion on this?

File: deimos/linenoise.di [hunk 1]
```diff
@@ -0,0 +1,61 @@
+/* linenoise.h -- guerrilla line editing library against the idea that a
```
Comment (MartinNowak) [inline deimos/linenoise.di]: Why do we want to copy that file here instead of using the deimos project as dependency?

**Comment by wilzbach** [comment]
OK moved to https://github.com/D-Programming-Deimos/linenoise/pull/5
With that, only a simple git tag should be needed :)

**Pull request merged**